topics/command-tips.md

```diff
@@ -0,0 +1,88 @@
+# Command tips
+
+An @array-reply of strings, command tips are meant to provide additional information about commands for clients and proxies.
+This infromation should help clients and proxies to determine how to handle this command when Redis is in clustering mode.
+Unlike the command flags (the third element in `COMMAND`'s reply), which are internal in nature (Redis itself uses them internally), command tips are just exposed via `COMMAND` and Redis does not use them.
+
+The tips are arbirary strings, but in order to create some sort of concensus, here is a list of suggested tips:
+
+## Boolean tips
+
+Tips that either exist or not, no value is associated with them.
+
+- **blocking**
+- **random-output**
+- **random-order-output**
+
+### blocking
+
+If present, this command has the potential to block (but not always, see `block_keyword` below)
+
+### random-output
+
+The output is not deterministic, that is, the same command with the same arguments, with the same key-space, may have different results. (e.g. `RANDOMKEY`, `SPOP`)
+
+### random-order-output
+
+The output is deterministic, but comes in random order (e.g. `HGETALL`, `SMEMBERS`)
+
+## Key-value tips
+
+Tips that are associated with a value.
+
+- **block_keyword** 
+- **request_policy**
+- **reply_policy**
+
+### block_keyword
+
+Identify commands which the potential to block, only is a specific arguement was given: For example, `XREAD` and `XREADGROUP` have the `blocking` tip, but they will only block if `BLOCK` was given. We will add the `block_keyword:<word>` tip, and the proxy should search `<word>` search in `argv` to determine if the command may indeed block.
+This could result in false-positive if that keyword is a keyname or some other form of free-text, but this tip best-effort: there's little harm in mistaking a non-blocking command with a blocking one (but mistaking in the other direction is unacceptable).
```

Comment:
Are we confident there's always little harm in false positive id of a blocking command?

I'm not too sure about that. For example, if a client will use this information to manage connections, such a false positive may lead to a connection pool exhausted unexpectedly only because specific key names / values were used.

Reply:
good point, but do you have suggestions for improvement? do something like key-specs?
FYI it's only relevant for XREAD and XREADGROUP, where the only opportunity for a rouge "block" is the keyname. maybe it's enough just to mention that we need to search argv from the beginning, ignoring args which are keys?

Reply:
did this design come from the RL proxy?
are there any modules that we're aware of that implement such an argument?
anyway, if we can't agree on it, i'm willing to leave it out of the scope for now, we can always add it later.

unless we wanna give an example of a tip in which part of the tip is a non-const argument for that tip

Reply:
@yossigo it's up to you to decide.
either we keep this flag and document the limitation
i.e. don’y have any non-key free text before the optional blocking keyword.
or we drop it for now, and just mark these commands as potentially blocking..
let me know what you prefer.

Reply:
BTW, this is related to this request: https://github.com/redis/redis/issues/3642

Reply:
we decided to drop it.

Reply:
removed block_keyword